test/reader.php

```diff
@@ -9,6 +9,7 @@
 $numberOfReads = 0;
 while (microtime(true) - $start < 3) {
     if (file_exists($file)) {
+        /** @var mixed $data */
```

Comment:
If the row below is replaced with an assertion, the suppression is no longer needed

Reply:
not sure what you mean here, I've tried:

```php
if (file_exists($file)) {
        $data = include $file;
        if (! assert(is_array($data))) { // <-- added `assert` here
            exit(1);
        }
```

but it does not work, still getting:
```
MixedAssignment - test/reader.php:12:9 - Unable to determine the type that $data is being assigned to (see https://psalm.dev/032)
```

The problem is that here we can have "mixed" value - in case reading file fails it is not an array.

Reply:
`assert()` already fails when running with assertions on.

Alternatively,. using an assertion library (I think phpunit's has the required `@psalm-assert` annotations)